src/js/jsx/tools/GuidesOverlay.jsx

```diff
@@ -60,17 +61,27 @@ define(function (require, exports, module) {
             var flux = this.getFlux(),
                 applicationStore = flux.store("application"),
                 toolStore = flux.store("tool"),
+                uiStore = flux.store("ui"),
                 modalState = toolStore.getModalToolState(),
+                uiState = uiStore.getState(),
                 currentTool = toolStore.getCurrentTool(),
                 currentDocument = applicationStore.getCurrentDocument();
 
             return {
                 document: currentDocument,
                 tool: currentTool,
-                modalState: modalState
+                modalState: modalState,
+                uiState: uiState
```

Comment:
Because of the React/D3 dance... I want any ui store and tool modal state changes to cause redraw (scrolling, zooming etc. all change window to canvas transform matrices, and we don't wanna show these during modal states)

Look at drawOverlay method for modalState being used, and any uiStore. functions within the file for how the ui store state is being used.